inc/report.class.php

```diff
@@ -304,72 +304,144 @@ static function showDefaultReport() {
+      array $order = [],
+      $extra = ''
+   ) {
       global $DB;
 
       // This SQL request matches the NetworkPort, then its NetworkName and IPAddreses. It also
       //      match opposite NetworkPort, then its NetworkName and IPAddresses.
       // Results are groupes by NetworkPort. Then all IPs are concatenated by comma as separator.
-      $query = "SELECT $field
-                       PORT_1.`itemtype` AS itemtype_1,
-                       PORT_1.`items_id` AS items_id_1,
-                       PORT_1.`id` AS id_1,
-                       PORT_1.`name` AS port_1,
-                       PORT_1.`mac` AS mac_1,
-                       PORT_1.`logical_number` AS logical_1,
-                       GROUP_CONCAT(ADDR_1.`name` SEPARATOR ',') AS ip_1,
-                       PORT_2.`itemtype` AS itemtype_2,
-                       PORT_2.`items_id` AS items_id_2,
-                       PORT_2.`id` AS id_2,
-                       PORT_2.`name` AS port_2,
-                       PORT_2.`mac` AS mac_2,
-                       GROUP_CONCAT(ADDR_2.`name` SEPARATOR ',') AS ip_2
-                FROM $networkport_prefix
-                INNER JOIN `glpi_networkports` AS PORT_1
-                    ON ($networkport_crit
-                        AND PORT_1.`is_deleted` = 0)
-                LEFT JOIN `glpi_networknames` AS NAME_1
-                    ON (NAME_1.`itemtype` = 'NetworkPort'
-                        AND PORT_1.`id` = NAME_1.`items_id`
-                        AND NAME_1.`is_deleted` = 0)
-                LEFT JOIN `glpi_ipaddresses` AS ADDR_1
-                    ON (ADDR_1.`itemtype` = 'NetworkName'
-                        AND NAME_1.`id` = ADDR_1.`items_id`
-                        AND ADDR_1.`is_deleted` = 0)
-                LEFT JOIN `glpi_networkports_networkports` AS LINK
-                    ON (LINK.`networkports_id_1` = PORT_1.`id`
-                        OR LINK.`networkports_id_2` = PORT_1.`id`)
-                LEFT JOIN `glpi_networkports` AS PORT_2
-                    ON (PORT_2.`id` = IF(LINK.`networkports_id_1` = PORT_1.`id`,
-                                         LINK.`networkports_id_2`,
-                                         LINK.`networkports_id_1`))
-                LEFT JOIN `glpi_networknames` AS NAME_2
-                    ON (NAME_2.`itemtype` = 'NetworkPort'
-                        AND PORT_2.`id` = NAME_2.`items_id`
-                        AND NAME_2.`is_deleted` = 0)
-                LEFT JOIN `glpi_ipaddresses` AS ADDR_2
-                    ON (ADDR_2.`itemtype` = 'NetworkName'
-                        AND NAME_2.`id` = ADDR_2.`items_id`
-                        AND ADDR_2.`is_deleted` = 0)
-                WHERE $where_crit
-                GROUP BY PORT_1.`id`";
-
-      if (!empty($order)) {
-         $query .= "ORDER BY $order";
+
+      if (count($joincrit) === 3) {
+         $andcrit = array_pop($joincrit);
+         $andcrit['AND']['PORT_1.is_deleted'] = 0;
```

Comment:
You could also add this into the WHERE clause, it may be simplier and could prevent some unwanted altering of `$joincrit`.